godot-core/src/builtin/packed_array.rs

```diff
@@ -0,0 +1,544 @@
+            /// elements after the removed element. The larger the array, the slower `remove` will be.
+            ///
+            /// # Panics
+            ///
+            /// If `index` is out of bounds.
+            // Design note: This returns the removed value instead of `()` for consistency with
+            // `Array` and with `Vec::remove`. Compared to shifting all the subsequent array
+            // elements to their new position, the overhead of retrieving this element is trivial.
+            pub fn remove(&mut self, index: usize) -> $Element {
+                self.check_bounds(index);
+                let element = self.get(index);
+                self.as_inner().remove_at(to_i64(index));
+                element
+            }
+
+            /// Assigns the given value to all elements in the array. This can be used together
+            /// with `resize` to create an array with a given size and initialized elements.
+            pub fn fill(&mut self, value: $Element) {
+                self.as_inner().fill(Self::into_arg(value));
+            }
+
+            /// Appends another array at the end of this array. Equivalent of `append_array` in
+            /// GDScript.
+            pub fn extend_array(&mut self, other: &$PackedArray) {
+                self.as_inner().append_array(other.clone());
+            }
+
+            /// Reverses the order of the elements in the array.
+            pub fn reverse(&mut self) {
+                self.as_inner().reverse();
+            }
+
+            /// Sorts the elements of the array in ascending order.
+            // Presumably, just like `Array`, this is not a stable sort so we might call it
+            // `sort_unstable`. But Packed*Array elements that compare equal are always identical,
+            // so it doesn't matter.
+            pub fn sort(&mut self) {
+                self.as_inner().sort();
+            }
+
+            /// Asserts that the given index refers to an existing element.
+            ///
+            /// # Panics
+            ///
+            /// If `index` is out of bounds.
+            fn check_bounds(&self, index: usize) {
+                let len = self.len();
+                assert!(
+                    index < len,
+                    "Array index {} is out of bounds: length is {}",
+                    index,
+                    len
+                );
+            }
+
+            /// Returns a pointer to the element at the given index.
+            ///
+            /// # Panics
+            ///
+            /// If `index` is out of bounds.
+            fn ptr(&self, index: usize) -> *const $Element {
+                self.check_bounds(index);
+                // SAFETY: We just checked that the index is not out of bounds.
+                let ptr = unsafe {
+                    (interface_fn!($operator_index_const))(self.sys(), to_i64(index)) as *const $Element
```

Comment:
Maybe also here, split statements into FFI call and pointer cast.
Explicit type annotations for the pointer is currently not possible, but can easily be added once we have generics.

Reply:
The type is either `$Element` or `TagType` or `TagString`. Had to add another macro argument for this, and it'll couple the implementation more tightly to the generated code. Which, I suppose, is exactly the idea :)